lib/acl-checker.js

```diff
@@ -81,127 +83,66 @@ class ACLChecker {
       })
   }
 
-  findAgentClass (graph, user, mode, resource, acl, callback) {
-    const debug = this.debug
-    // Agent class statement
-    var agentClassStatements = graph.match(acl,
-      'http://www.w3.org/ns/auth/acl#agentClass')
-    if (agentClassStatements.length === 0) {
-      return callback(false)
-    }
-    async.some(
-      agentClassStatements,
-      function (agentClassTriple, found) {
-        // Check for FOAF groups
-        debug('Found agentClass policy')
-        if (agentClassTriple.object.uri === 'http://xmlns.com/foaf/0.1/Agent') {
-          debug(mode + ' allowed access as FOAF agent')
-          return found(true)
-        }
-        return found(false)
-      },
-      callback)
-  }
-
+  /**
+   * @method findRule
+   * @param graph {Graph} Parsed RDF graph of current .acl resource
+   * @param user {String} WebID URI of the user accessing the resource
+   * @param mode {String} Access mode, e.g. 'Read', 'Write', etc.
+   * @param resource {String} URI of the resource being accessed
+   * @param accessType {String} One of `accessTo`, or `default`
+   * @param acl {String} URI of this current .acl resource
+   * @param callback {Function}
+   * @param options {Object} Options hashmap
+   * @param [options.origin] Request's `Origin:` header
+   * @param [options.host] Request's host URI (with protocol)
+   */
   findRule (graph, user, mode, resource, accessType, acl, callback, options) {
     const debug = this.debug
     if (!graph || graph.length === 0) {
       debug('ACL ' + acl + ' is empty')
-      return callback(new Error('No policy found'))
+      return callback(new Error('No policy found - empty ACL'))
     }
-    debug('Found policies in ' + acl)
-    // Check for mode
-    var statements = this.getMode(graph, mode)
-    if (mode === 'Append') {
-      statements = statements.concat(this.getMode(graph, 'Write'))
+    let isContainer = accessType.startsWith('default')
+    let aclOptions = {
+      aclSuffix: this.suffix,
+      graph: graph,
+      host: options.host,
```

Comment:
`host` gets set in the `allow` handler, [here](https://github.com/solid/node-solid-server/blob/80e0675459a69a2fd25f2a6eb804a5dee05a7e84/lib/handlers/allow.js#L42).
It's used by the 'enforce strict origin' code in the permissions lib.